File: 코딩테스트/이진탐색/sol-1.py
import sys

n = sys.stdin.readline().rstrip()
array1 = list(map(int, sys.stdin.readline().rstrip().split()))
array1.sort()

m = sys.stdin.readline().rstrip()
array2 = list(map(int, sys.stdin.readline().rstrip().split()))

def binarySearch(array, target, start, end):
    if start > end:
        return None
    mid = (start + end) // 2

    if array[mid] == target:
        return mid
    elif array[mid] > target:
        return binarySearch(array, target, start, mid - 1)
    else:
        return binarySearch(array, target, mid + 1, end)


for i in range(len(array2)):
    result = binarySearch(array1, array2[i], 0, len(array1) - 1)
    if result == None:
        print('no', end=' ')
    else:
        print('yes', end=' ')

File: 코딩테스트/이진탐색/sol-1.js
function solution(N, stock, M, order) {
  const sortedStock = stock.sort();
  const sortedOrder = order.sort();

  for (const item of sortedOrder) {
    const result = binarySearch(sortedStock, Number(item), 0, N - 1);
    if (result) {
      console.log("yes");
    } else {
      console.log("no");
    }
  }
}

function binarySearch(array, target, start, end) {
  while (start <= end) {
    const mid = Math.floor((start + end) / 2);
    if (Number(array[mid]) === target) {
      return mid;
    }
    if (Number(array[mid]) > target) {
      end = mid - 1;
    } else {
      start = mid + 1;
    }
  }
  return false;
}

const N = 5;
const stock = [8, 3, 7, 9, 2];
const M = 3;
const order = [5, 7, 9];

solution(N, stock, M, order);
